tethysext/atcore/controllers/resource_workflows/results_views/report_workflow_results_view.py

```diff
@@ -0,0 +1,221 @@
+                        legend_info = None
+                        if params:
+                            if 'TILED' in params.keys():
+                                params.pop('TILED')
+                            if 'TILESORIGIN' in params.keys():
+                                params.pop('TILESORIGIN')
+                            result_layer['options']['params'] = params
+
+                            legend_key = result_layer['data']['layer_id']
+                            if ":" in legend_key:
+                                legend_key = legend_key.replace(":", "_")
+                            legend_info = {
+                                'legend_id': legend_key,
+                                'title': result_layer['legend_title'].replace("_", " "),
+                                'units': result.options['units'] if 'units' in result.options.keys() else "",
+                                'divisions': dict()
+                            }
+
+                            # Uses param ENV to create the scale
+                            divisions = params['ENV'].split(";")
+                            divisions_dict = {}
+
+                            for division in divisions:
+                                division = division.split(":")
+                                divisions_dict[division[0]] = division[1]
+
+                            for k, v in divisions_dict.items():
+                                if 'val' in k and k[:11] != 'val_no_data':
+                                    legend_info['divisions'][float(v)] = divisions_dict[k.replace('val', 'color')]
+                            legend_info['divisions'] = OrderedDict(
+                                sorted(legend_info['divisions'].items())
+                            )
+                        result_layer.options['url'] = self.geoserver_url(result_layer.options['url'])
+                        # Add layer to beginning the map's of layer list
+                        map_view.layers.insert(0, result_layer)
+                        # Append to final results list.
+                        results.append({'map': [result.name, result.description, legend_info, result_layer]})
+
+        # Save changes to map view and layer groups
+        context.update({
+            'can_run_workflows': can_run_workflows,
+            'has_tabular_data': has_tabular_data,
+            'tabular_data': tabular_data,
+            'report_results': results,
+        })
+        # Note: new layer created by super().process_step_options will have feature selection enabled by default
+        super().process_step_options(
+            request=request,
+            session=session,
+            context=context,
+            resource=resource,
+            current_step=current_step,
+            previous_step=previous_step,
+            next_step=next_step
+        )
+
+    def get_context(self, request, session, resource, context, model_db, workflow_id, step_id, result_id, *args,
```

Comment:
Don't need this method if not inheriting from both MapWorkflowView and ResultWorkflowView (see previous comment).